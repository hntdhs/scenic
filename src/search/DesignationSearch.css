/* .link {
    padding-left: 5em;
} */