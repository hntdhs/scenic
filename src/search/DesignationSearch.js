import React, { useEffect, useState } from "react";
import Checkbox from "./Checkbox";
import "./DesignationSearch.css";


function DesignationSearch({ onChange }) {
    const categories = ['All-American Road', 'National Scenic Byway'];
    const [selected, setSelected] = useState([]);

    useEffect(() => {
        onChange(selected)
    }, [selected]);

    const changeHandler = (props) => {
        if (props.checked) {
            const index = selected.indexOf(props.value)
            if (index > -1) {
                selected.splice(index, 1)
                setSelected(selected)
            } else {
                selected.push(props.value);
                setSelected(selected)
            }
            // this is used in SearchForm so that's where the value prop is given - value={geoFeaturesSelect}
            
        }
    }

        return (
            <div>
                <div className="link">
                    <a href="/designations-explanation">What are these?</a>
                </div>
                {categories.map((c, i) => (
                    <div key={i} className="form-check form-check-inline">
                        <label key={c} htmlFor={c}>{c}</label>
                        <Checkbox onChange={changeHandler} name={c}/>
                    </div>
                ))
        
                }
            </div>
        )
}

export default DesignationSearch;