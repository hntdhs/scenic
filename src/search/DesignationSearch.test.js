import React from "react";
import { render } from "@testing-library/react";
import DesignationSearch from "./DesignationSearch.js";
import { expect } from "@jest/globals";

it("matches snapshot", function () {
  const testHandlerMock = jest.fn(x => 42 + x);
    const { asFragment } = render(<DesignationSearch onChange={testHandlerMock} />);
    expect(asFragment()).toMatchSnapshot();
    expect(testHandlerMock).toBeCalled()
  });