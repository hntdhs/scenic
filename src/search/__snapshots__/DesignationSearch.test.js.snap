// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="link"
    >
      <a
        href="/designations-explanation"
      >
        What are these?
      </a>
    </div>
    <div
      class="form-check form-check-inline"
    >
      <div>
        <input
          class="form-check-input"
          id="flexCheckDefault"
          name="All-American Road"
          type="checkbox"
          value="All-American Road"
        />
        <label>
          All-American Road
        </label>
      </div>
    </div>
    <div
      class="form-check form-check-inline"
    >
      <div>
        <input
          class="form-check-input"
          id="flexCheckDefault"
          name="National Scenic Byway"
          type="checkbox"
          value="National Scenic Byway"
        />
        <label>
          National Scenic Byway
        </label>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
